dev_tools/scripts/create_new_algorithm.sh

```diff
@@ -146,131 +154,146 @@ TEMPLATE_FILES_DESCRIPTIONS=(
-    perl -pi -e "s/${source_name^^}/$NAME/g" ${TMPDIR}/$destpath
+    # Evaluate the extension and try to find the matching files
+    for j in $(ls ${GINKGO_ROOT_DIR}/${sourcepath})
+    do
+        if [ -f "$j" ]
+        then
+            filename=$(basename -- ${j})
+            source_path=${TEMPLATE_FILES_LOCATIONS[$i-1]}/${filename}
+            destname=$(echo "${filename}" | sed "s/${source_name}/${name}/")
+            destpath=${TEMPLATE_FILES_LOCATIONS[$i-1]}/$destname
 
+            cp ${GINKGO_ROOT_DIR}/$source_path ${TMPDIR}/$destpath
 
-    # Comment all code
-    awk '/^{$/,/^}$/ { if ($0 == "{"){ print "GKO_NOT_IMPLEMENTED;"; print "//" $0; print "// TODO (script): change the code imported from '${source_type}'/'${source_name}' if needed"; next} else { print "//" $0; next }} 1' ${TMPDIR}/$destpath > tmp
-    mv tmp  ${TMPDIR}/$destpath
+            # Replace all instances of source_name by the user's requested name
+            perl -n -i -e "print unless m/.*common.*${source_name}_kernels.hpp.inc.*/" ${TMPDIR}/$destpath
+            perl -pi -e "s/${source_name}/$name/g" ${TMPDIR}/$destpath
+            perl -pi -e "s/${source_name^}/$Name/g" ${TMPDIR}/$destpath
 
-    ls ${TMPDIR}/$destpath
-done
+            # Comment all code
+            awk -v name=${name} '/^{$/,/^}$/ { if ($0 == "{"){ print "GKO_NOT_IMPLEMENTED;"; print "//" $0; print "// TODO (script:" name "): change the code imported from '${source_type}'/'${source_name}' if needed"; next} else { print "//" $0; next }} 1' ${TMPDIR}/$destpath > tmp
+            mv tmp ${TMPDIR}/$destpath
 
-if [ $execute == 1 ]
-then
-    echo -e "\nRenaming and distributing files"
-    # rename and distribute the files to the right location
-    # for each file, make sure it does not exist yet
-    for (( i=1; i<${#TEMPLATE_FILES[@]}+1; i++ ))
-    do
-        sourcepath=${TEMPLATE_FILES_LOCATIONS[$i-1]}/${TEMPLATE_FILES[$i-1]}
-        destpath=${TEMPLATE_FILES_LOCATIONS[$i-1]}/${TEMPLATE_FILES[$i-1]}
-        if [ ! -f ${GINKGO_ROOT_DIR}/$destpath ]; then
-            cp ${TMPDIR}/${sourcepath} ${GINKGO_ROOT_DIR}/${destpath}
+            ls ${TMPDIR}/$destpath
+
+            if [ $execute == 1 ]
+            then
+                if [ ! -f ${GINKGO_ROOT_DIR}/$destpath ]; then
+                    cp ${TMPDIR}/${destpath} ${GINKGO_ROOT_DIR}/${destpath}
+                else
+                    echo -e "Error: file ${GINKGO_ROOT_DIR}/$destpath exists"
+                    echo -e "Remove file first if you want to replace it."
+                    read -p ""
+                fi
+            fi
         else
-            echo -e "Error: file ${GINKGO_ROOT_DIR}/$destpath exists"
-            echo -e "Remove file first if you want to replace it."
-            read -p ""
+            echo "Warning: Source file $sourcepath was not found."
         fi
     done
+done
 
-
+if [ $execute == 1 ]
+then
     echo -e "cleaning up temporary files."
     rm -rf ${TMPDIR}
```

Comment:
On my Mac, this causes problems as if you remove the ${TMPDIR}, then `mktemp` in line 283 does not copy/modify the CMakeLists.txt properly. If I move this removal of the ${TMPDIR} while exiting the if clause, then everything works as expected.

Reply:
Thanks for checking. There is no problem with the `grep` command? Has Mac has different bash tools, I was a bit unsure about this.

Reply:
No, `grep` seems to work fine for me. I have `grep` from BSD which I think has full compatibility with GNU grep.

Reply:
Could you try reverting to the default one and check if this works? Or is that the default one?

Reply:
I think the one I have is the default one you get with Mac.

Reply:
I don't understand why the `${TMPDIR}` has anything to do with `mktemp`, as they aren't used togother and do `${TMPDIR}` is not made from `mktemp` actually. I put the `rm` at the end, how is it now?

Reply:
I say the grep works fine because, for example in the `hip/CMakeLists.txt`, the file is added correctly to the `GINKGO_HIP_SOURCES` and the other `CMakeLists.txt` additions are correct as well.

Reply:
About grep, I also meant the last one which finds automatically generates a list of files with lines of TODO items.

Reply:
Yes, that seems to be correct as well. I do get a huge list of TODO with line numbers and file names.

Reply:
OK thanks for checking